File: leetcode/rust-leetcode/src/724.find-pivot-index.rs
impl Solution {
    pub fn pivot_index(nums: Vec<i32>) -> i32 {
        let sum:i32 = nums.iter().sum();
        let mut leftsum:i32 = 0;
        for (i, n) in nums.iter().enumerate() {
            if leftsum == sum - n - leftsum {
                return i as i32;
            }
            leftsum += n;
        }
        -1
    }
}

File: leetcode/python/724.find-pivot-index.py
class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        s = sum(nums)
        cs = 0
        for i in range(len(nums)):
            if cs * 2 + nums[i] == s:
                return i
            cs += nums[i]
        return -1

